### PR TITLE
Support generic POJO super classes

### PR DESCRIPTION
**Pull request by eranl**
Also, simplify type checking.

Should I add this to the manual?

### CODE REVIEW

**Comment by majst01** [comment]
Support for Generics is for sure worth mentioning in the manual.

**Comment by eranl** [comment]
> Support for Generics is for sure worth mentioning in the manual.

Done

File: src/main/java/org/influxdb/impl/TypeMapper.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        throw new IllegalStateException("Could not resolve type " + type);
+      }
+
+      return resolvedType;
+    }
+
+    return type;
+  }
+
+  static Type map(TypeVariable<?> typeVariable, TypeMapper innerMapper, TypeMapper outerMapper) {
```
Comment (majst01) [inline src/main/java/org/influxdb/impl/TypeMapper.java:73]: This function is not covered in tests, is it even used ?

Comment (eranl) [reply src/main/java/org/influxdb/impl/TypeMapper.java]: Sorry, this was a leftover. Deleted. Thanks for the catch!

**Review by majst01 — APPROVED** [review]

**Pull request merged**